source/http_stream.c

```diff
@@ -121,7 +121,7 @@ static int s_on_incoming_header_block_done(
             goto done;
         }
 
-        PyList_SET_ITEM(header_list, i, tuple); /* steals reference to tuple */
+        PyList_SetItem(header_list, i, tuple); /* steals reference to tuple */
```

Comment:
this doesn't "steal a reference to the the tuple" anymore so this is a leak now

see, this is why it's good to comment on the lifetime stuff. If this comment wasn't here I wouldn't have remembered that